### ISSUE TITLE
training video ram. and some warning 

### ISSUE THREAD

**Issue by wildbrother** [zylo117/Yet-Another-EfficientDet-Pytorch#658]
before issue .  I upgraded my torch to newest. so the momory issue is solved. 
but the warning is now occurred. (Q2).

Q1.
when i run d1 model train . batch_size=4 , with 4 GPU //
the gpu memory is 3200mb
![Tue Jun 15 172849 2021](https://user-images.githubusercontent.com/10546211/122269641-6aa07f00-cf18-11eb-9e0e-8ac79c2f44f5.png)

gpu 4 , batchsize 4 -> one batch for 1 gpu.
and, I can't train D6 model in batch:4 with RTX TITAN X 4

Is it normal memory usage in training??

Q2.
when i upgrade my torch to 1.8.1
I got a message in Val.Epoche // I'd never get an message in this phase, when my torch ver. is 1.4.1

[W accumulate_grad.h:184] Warning: grad and param do not obey the gradient layout contract. This is not an error, but may impair performance.

![](https://user-images.githubusercontent.com/10546211/122023012-1d31ee00-ce02-11eb-9e2a-2cd188777de2.png)

is it critical bug..?

**Comment by zylo117** [comment]
1. yes, seems normal
2. I havn't met it yet. Did you modify the code?

**Comment by wildbrother** [comment]
> * yes, seems normal
> * I havn't met it yet. Did you modify the code?

1. Do you think this memory usage will decrease dramatically if we use tensorflow? (static graph)

2. no I didn't modify the model code. the warning code said, gradient layout is wrong.. but i didn't modify that.  before pytorch version I didn't met that warning.

**Comment by wildbrother** [comment]
3. I will train D0~D3 model.  and inference this model with larger input size (like the official and your inference picture) 
is it good strategy for lifting up map? 
// because I don't really have much bigger memory ram to train bigger model. but in inference there are idle gpu memory. so i want to use this idle memory

**Comment by zylo117** [comment]
1. no. maybe by a little bit
2. try torch+cuda111 version.
3. train d0 in larger input size can get better map

**Comment by wildbrother** [comment]
> * no. maybe by a little bit
> * try torch+cuda111 version.
> * train d0 in larger input size can get better map

oh.. small feature map & high input resolution makes better MAP ?
It is not match with the paper (the Compound Scailing)   
do you think training input resolution is critical than feature map capacity?

ok.. 
you suggest me 
1. train with high resolution input (with small model).  
ok  now I'm trying D1 with 1500 input resolution

how about inference with high resolution?

**Comment by zylo117** [comment]
I have never said small feature map.
I said larger input size, hence larger feature map.

of course it consumes more memory

And inference with high resolution doesn't always help
